drivers/SmartThings/matter-sensor/src/air-quality-sensor/init.lua

```diff
@@ -164,130 +164,103 @@ local supported_profiles =
-  end
-  if #tvoc_level_eps > 0 then
-    level_indication_support = level_indication_support .. "-tvoc"
-  end
+
+  local meas_name, level_name = create_level_measurement_profile(device)
+
   -- If all endpoints are supported, use '-all' in the profile name so that it
   -- remains under the profile name character limit
-  if level_indication_support == "-co-co2-no2-ozone-formaldehyde-pm1-pm25-pm10-radon-tvoc" then
-    level_indication_support = "-all"
+  if level_name == "-co-co2-no2-ozone-ch2o-pm1-pm25-pm10-radon-tvoc" then
+    level_name = "-all"
   end
-  if level_indication_support ~= "" then
-    profile_name = profile_name .. level_indication_support .. "-level"
+  if level_name ~= "" then
+    profile_name = profile_name .. level_name .. "-level"
   end
 
-  if #co_meas_eps > 0 then
-    numeric_measurement_support = numeric_measurement_support .. "-co"
-  end
-  if #co2_meas_eps > 0 then
-    numeric_measurement_support = numeric_measurement_support .. "-co2"
-  end
-  if #no2_meas_eps > 0 then
-    numeric_measurement_support = numeric_measurement_support .. "-no2"
-  end
-  if #ozone_meas_eps > 0 then
-    numeric_measurement_support = numeric_measurement_support .. "-ozone"
-  end
-  if #formaldehyde_meas_eps > 0 then
-    numeric_measurement_support = numeric_measurement_support .. "-formaldehyde"
-  end
-  if #pm1_meas_eps > 0 then
-    numeric_measurement_support = numeric_measurement_support .. "-pm1"
-  end
-  if #pm2_5_meas_eps > 0 then
-    numeric_measurement_support = numeric_measurement_support .. "-pm25"
-  end
-  if #pm10_meas_eps > 0 then
-    numeric_measurement_support = numeric_measurement_support .. "-pm10"
-  end
-  if #radon_meas_eps > 0 then
-    numeric_measurement_support = numeric_measurement_support .. "-radon"
-  end
-  if #tvoc_meas_eps > 0 then
-    numeric_measurement_support = numeric_measurement_support .. "-tvoc"
-  end
   -- If all endpoints are supported, use '-all' in the profile name so that it
   -- remains under the profile name character limit
-  if numeric_measurement_support == "-co-co2-no2-ozone-formaldehyde-pm1-pm25-pm10-radon-tvoc" then
-    numeric_measurement_support = "-all"
+  if meas_name == "-co-co2-no2-ozone-ch2o-pm1-pm25-pm10-radon-tvoc" then
+    meas_name = "-all"
   end
-  if numeric_measurement_support ~= "" then
-    profile_name = profile_name .. numeric_measurement_support .. "-meas"
+  if meas_name ~= "" then
+    profile_name = profile_name .. meas_name .. "-meas"
   end
 
   if not tbl_contains(supported_profiles, profile_name) then
     device.log.warn_with({hub_logs=true}, string.format("No matching profile for device. Tried to use profile %s", profile_name))
-    if #co_meas_eps > 0 or #no2_meas_eps > 0 or #ozone_meas_eps > 0 or #formaldehyde_meas_eps > 0 or
-        #pm1_meas_eps > 0 or #pm10_meas_eps > 0 or #radon_meas_eps > 0 then
+
+    local function meas_find(sub_name)
+      return string.match(meas_name, sub_name)
```

Comment:
You could optimize this to be
```
return string.match(meas_name, sub_name) ~= nil
```

Since we really only want to know if the substring is present, there's no need to return the found sub string.

Reply:
I like it :+1: